### PR TITLE
[Task #68029]: View product in stock

### PR DESCRIPTION
**Pull request by BuiMinhDung711**
## Related Tickets:
- [#68029](https://edu-redmine.sun-asterisk.vn/issues/68029)

## WHAT this PR does?

- Show user the product in stock

## Attachments:
image

### CODE REVIEW

**Comment by BuiMinhDung711** [comment]
ready

File: app/Http/Controllers/Auth/AuthenticatedSessionController.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function store(LoginRequest $request)
         $request->session()->regenerate();
 
         if (Auth::user()->hasRole('admin')) {
-            return redirect('/dashboard');
+            return redirect('/admin/dashboard');
```
Comment (ghost) [inline app/Http/Controllers/Auth/AuthenticatedSessionController.php]: use route()

File: app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -210,18 +221,33 @@ public function update(UpdateProductRequest $request, Product $product)
      */
     public function destroy(Product $product)
     {
+        if ($product->deleted_at !== null) {
+            abort(403);
+            report('Product was deleted.');
```
Comment (ghost) [inline app/Http/Controllers/ProductController.php]: report ko có ý nghĩa, function break ngay sau abort :x xóa đi nha...

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -210,18 +221,33 @@ public function update(UpdateProductRequest $request, Product $product)
...
 
+    public function recreate(Product $product)
+    {
+        if ($product->deleted_at == null) {
+            abort(403);
+            report('Product was deleted.');
+        }
+        $product->deleted_at = null;
+        $product->save();
+        return redirect()->route('product.edit', $product)->with('status', 'product-recreated');
```
Comment (ghost) [inline app/Http/Controllers/ProductController.php]:
```suggestion
        $product->save();

        return redirect()->route('product.edit', $product)->with('status', 'product-recreated');
```

File: app/Http/Controllers/ProductInStockController.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  \App\Models\ProductInStock  $productInStock
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(ProductInStock $productInStock)
+    {
+        $productInStock->delete();
+        return back();
```
Comment (ghost) [inline app/Http/Controllers/ProductInStockController.php]:
```suggestion
        $productInStock->delete();

        return back();
```

File: app/Http/Livewire/ProductDetailView.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+use App\Models\Product;
+use LaravelViews\Facades\UI;
+use LaravelViews\Views\DetailView;
+
+class ProductDetailView extends DetailView
+{
+    protected $modelClass = Product::class;
+    public $title = "Product";
+    public $subtitle = "The details of the product";
```
Comment (ghost) [inline app/Http/Livewire/ProductDetailView.php]: localization?

File: app/Http/Livewire/ProductDetailView.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    public function heading($model)
+    {
+        return [
+            "Product: {$model->name}",
+            "This is the details of {$model->name}",
```
Comment (ghost) [inline app/Http/Livewire/ProductDetailView.php]: localization

File: app/View/Actions/ShowProductAction.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+use LaravelViews\Actions\Action;
+use LaravelViews\Views\View;
+
+class ShowProductAction extends Action
+{
+    /**
+     * Any title you want to be displayed
+     * @var String
+     * */
+    public $title = "Product's details";
```
Comment (ghost) [inline app/View/Actions/ShowProductAction.php]: localization

**Comment by BuiMinhDung711** [comment]
ready

File: app/Http/Controllers/ProductController.php [hunk 8]
```diff
@@ -210,18 +221,34 @@ public function update(UpdateProductRequest $request, Product $product)
...
         $product->save();
+        
         return redirect()->route('product.edit', $product)->with('status', 'product-deleted');
     }
 
+    public function recreate(Product $product)
+    {
+        if ($product->deleted_at == null) {
+            abort(403);
+            report('Product was deleted.');
```
Comment (ghost) [inline app/Http/Controllers/ProductController.php]: function stop ở abort... report ko có ý nghĩa nhỉ

File: app/Http/Livewire/ProductDetailView.php (hunk 6 above)
Comment (ghost) [reply app/Http/Livewire/ProductDetailView.php]: chua fix

File: app/View/Actions/ShowProductAction.php (hunk 7 above)
Comment (ghost) [reply app/View/Actions/ShowProductAction.php]: chua fix

**Comment by BuiMinhDung711** [comment]
ready

**Pull request merged**